test/unit/components/navpatientheader/index.test.js

```diff
@@ -0,0 +1,298 @@
+/* global chai */
+/* global describe */
+/* global sinon */
+/* global afterEach */
+/* global context */
+/* global it */
+/* global beforeEach */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { BrowserRouter } from 'react-router-dom';
+import _ from 'lodash';
+
+import NavPatientHeader from '../../../../app/components/navpatientheader';
+
+const expect = chai.expect;
+
+const defaultPatientProps = { 
+  userid: 1234,
+  profile: { 
+    fullName: 'Vasyl Lomachenko',
+    patient: { birthday: '2010-10-20', mrn: '567890' }
+  }
+};
+
+describe('NavPatientHeader', () => {
+  const mockHistory = { push: sinon.stub() }
+  const mockTrackMetric = sinon.stub();
+
+  NavPatientHeader.__Rewire__('useHistory', sinon.stub().returns(mockHistory));
+
+  afterEach(() => {
+    mockTrackMetric.reset();
+    mockHistory.push.reset();
+  });
+
+  describe('visibility of patient info and actions', () => {
+    context('personal user with non-root permissions', () => {
+      const props = { // personal user with non-root permissions;
+        user: { roles: [] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: false } },
```

Comment:
A quirk of the `permissions` object is that the permissions are "true" if the property is present and is an empty object, so `permissions: { root: {} }` for true and just `permissions: {}` for false.

Reply:
Interesting, yeah I see it now. This one will take a little getting used to.
![image](https://github.com/user-attachments/assets/ffa0f092-f035-453b-92dc-91af0eb2e608)

dad2cb888f30181c613320a650b2e1778d46ed05
6414083b7c697b2dc061aed862f8def6ed6951e3